lib/engine/game/g_18_jpt/game.rb

```diff
@@ -269,13 +267,14 @@ def tile_lays(entity)
         end
 
         def upgrades_to?(from, to, _special = false, selected_company: nil)
-          if from.color == :green
-            return to.name == 'J7' if from.hex.name == 'D96'
-            return to.name == 'J14' if %w[E77 F94].include?(from.hex.name)
-            return to.name == 'J15' if from.hex.name == 'J88'
-          end
+          # Allow merging of two separate cities into one with two slots
+          return to.name == '611' if from.hex.coordinates == 'F92' && from.color == :green
 
-          return false if RESERVED_TILES.include?(to.name)
+          # KU hexes have default upgrades in all colors but brown so check without label for simplicity
+          if %w[E77 F94].include?(from.hex.name) && from.color != :brown
+            from = from.dup
```

Comment:
what's the point of duping here?

Reply:
@tobymao these hexes have custom upgrades only from brown so there are two ways:
1) describe every possible legal upgrade here (eg write 6, 57 for white, 14,15 for yellow etc)
2) check for upgrade eligibility ignoring label on this hex

I chose second option. And as I don't know if `from` is a real tile that is on the map, or its duplicate, I dup it to avoid my label removal persisting to the map.

Reply:
i don't think it's doing what you're doing, dup is a shallow copy of from, if you do labels.clear, that's probably the same reference to the underlying list, are you user that this code is doing what you intend?

Reply:
Well, it seems to work as I intend it to, but I think I'll do more tests of you're concerned

Reply:
@tobymao done this in other way